src/query_parser/v2/parser.y

```diff
@@ -846,34 +835,16 @@ expr(A) ::= modifier(B) COLON geo_filter(C). {
   }
 }
 
-geo_filter(A) ::= LSQB param_any(B) param_any(C) param_any(D) param_any(E) RSQB. [NUMBER] {
-  // Update token type to be more specific if possible
-  // and detect syntax errors
-  QueryToken *badToken = NULL;
-
-  if (B.type == QT_PARAM_ANY)
+geo_filter(A) ::= LSQB param_num(B) param_num(C) param_num(D) param_term(E) RSQB. [NUMBER] {
```

Comment:
I have tried to add `geo_filter(A) ::= LSQB param_num(B) param_num(C) param_num(D) param_num(E)` and got a conflict. Can this PR help resolve it?

Reply:
No, this PR cannot enable that, as even we cannot tell what is the appropriate rule to choose in queries like `@g:[$p1 $p2 $p3 $p4]`